File: ROS (ROBOT OPERATING SYSTEM)/beginner_tutorials/src/colorchangesrv.cpp
#include <ros/ros.h>
#include <std_srvs/Empty.h>
#include <turtlesim/Spawn.h>

int main(int argc, char** argv)
{
	ros::init(argc, argv, "setcolor");
	ros::NodeHandle n;
	
	//int color_r = 255, color_b = 25, color_g = 20;
	int color_r, color_b, color_g;	

	n.getParam("r", color_r);
	n.getParam("g", color_g);
	n.getParam("b", color_b);

	ros::service::waitForService("clear");

	ros::param::set("background_r", color_r);
	ros::param::set("background_g", color_g);
	ros::param::set("background_b", color_b);

	ros::ServiceClient clearclient = n.serviceClient<std_srvs::Empty>("clear");
	std_srvs::Empty srv;

	clearclient.call(srv);
	
	ros::ServiceClient sclient = n.serviceClient<turtlesim::Spawn>("spawn");

	turtlesim::Spawn::Request req;
	turtlesim::Spawn::Response resp;

	req.x = 2;
	req.y = 2;
	req.theta = 0.5;
	req.name = "turtle2";

	bool success = sclient.call(req, resp);
}

File: ROS (ROBOT OPERATING SYSTEM)/beginner_tutorials/src/colorchangesrv.py
#! /usr/bin/env python

#Make the node an executable node
#chmod u+x ~/catkin_ws/src/beginner_tutorials/src/colorchangesrv.py

import rospy
from std_srvs.srv import Empty 
from turtlesim.srv import*

rospy.init_node('setcolor', anonymous=True)

color_r = rospy.get_param('r')
color_g = rospy.get_param('g')
color_b = rospy.get_param('b')

print(color_r,color_g,color_b)

rospy.wait_for_service("clear")

rospy.set_param("background_r",color_r)
rospy.set_param("background_g",color_g)
rospy.set_param("background_b",color_b)

clear_background = rospy.ServiceProxy('clear', Empty)
clear_background()

sturtle = rospy.ServiceProxy('spawn', Spawn)
sturtle(2,2,0.5,"turtle2")